spinoso-string/src/enc/utf8/mod.rs

```diff
@@ -0,0 +1,1060 @@
+mod eq;
+mod impls;
+
+use alloc::collections::TryReserveError;
+use alloc::vec::Vec;
+use core::fmt;
+use core::hash::{Hash, Hasher};
+use core::ops::Range;
+use core::slice::SliceIndex;
+
+use bstr::{ByteSlice, ByteVec};
+
+use crate::codepoints::InvalidCodepointError;
+use crate::iter::{Bytes, IntoIter, Iter, IterMut};
+use crate::ord::OrdError;
+
+#[allow(clippy::module_name_repetitions)]
+#[derive(Default, Clone)]
+pub struct Utf8String {
+    inner: Vec<u8>,
+}
+
+impl Hash for Utf8String {
+    fn hash<H: Hasher>(&self, hasher: &mut H) {
+        self.as_slice().hash(hasher);
+    }
+}
+
+impl PartialEq for Utf8String {
+    fn eq(&self, other: &Self) -> bool {
+        *self.inner.as_slice() == *other.inner.as_slice()
+    }
+}
+
+impl Eq for Utf8String {}
```

Comment:
```suggestion
}
```

Reply:
Oof, This was no doubt accidentally included. thanks for catching!